packages/sdk/src/messageExtension/executeWithSSO.ts

```diff
@@ -40,6 +79,64 @@ function getSignInResponseForMessageExtension(teamsfx: TeamsFx, scopes: string |
   };
 }
 
+/**
+ * execution in message extension with SSO token.
+ *
+ * @param {TurnContext} context - The context object for the current turn.
+ * @param {OnBehalfOfCredentialAuthConfig} authConfig - User custom the message extension authentication configuration.
+ * @param {initiateLoginEndpoint} initiateLoginEndpoint - Login page for Teams to redirect to.
+ * @param {string[]} scopes - The list of scopes for which the token will have access.
+ * @param {function} logic - Business logic when executing the query in message extension with SSO or access token.
+ *
+ * @throws {@link ErrorCode|InternalError} when failed to get access token with unknown error.
+ * @throws {@link ErrorCode|TokenExpiredError} when SSO token has already expired.
+ * @throws {@link ErrorCode|ServiceError} when failed to get access token from simple auth server.
+ * @throws {@link ErrorCode|InvalidParameter} when scopes is not a valid string or string array.
+ * @throws {@link ErrorCode|RuntimeNotSupported} when runtime is nodeJS.
+ *
+ * @returns A MessageExtension Response for the activity. If the logic not return any, return void instead.
+ */
+export async function executionWithTokenAndConfig(
```

Comment:
is it possible to merge executionWithTokenAndConfig and executionWithToken (e.g. using the way you take with botSsoConfig https://github.com/OfficeDev/TeamsFx/pull/6437/files?file-filters%5B%5D=.ts&show-viewed-files=true#diff-d239c7c664fd0f35ac9ad3a3d082d76b9f971445b4d195e20ab0adf78633e763R329)? My concern here is user might be confused about the difference  between them.
cc @wenytang-ms

Reply:
Do you mean this function `handleMessageExtensionQueryWithSSO`? Since executionWithTokenAndConfig is not export to user.

As discussed with PM, we plan to deprecate `handleMessageExtensionQuery` function in teamsfx 2.0, and remove it for split sdk, so we will use another function name to make the logic clearer and easy for future to remove.

Reply:
Yes, it's for `handleMessageExtensionQueryWithSSO` which is exposed to developers. Sorry for misleading. The reason to have this additional function seems to be an engineering reason, which should not impact user facing things in my own opinion.
You may keep this function and call it from `handleMessageExtensionQueryWithToken` to simplify our further development and avoid confusing users with 2 similar functions.
Another alternative is deprecate `handleMessageExtensionQueryWithToken` and tell users to use `handleMessageExtensionQueryWithSSO` in the deprecation message.
All I want to achieve to avoid possible confusions with 2 similar functions.
cc @adashen @MuyangAmigo

Reply:
Yes, we will add deprecate notice for the function `handleMessageExtensionQueryWithToken`, however, due to docs are not fully ready, user may see some deprecate notice but cannot find docs about how to use it, so we plan to add the notice for the next release.

Reply:
Thanks, I'm fine with the deprecate way.